### PR TITLE
Use unmodified command for the command history.

### PR DESCRIPTION
**Pull request by morethanoneanimal**
Closes awesomeWM/awesome#1104.

### CODE REVIEW

**Comment by Elv13** [comment]
Thanks a lot. I guess this could use a test, but then again, this file isn't well tested yet, so it might be a little too much to ask given there is some internal data that will need to be exposed. In turn, that will require and API talk and so on.

I will wait to see what @blueyed think about it. But really, thanks for attacking the blocking bugs list, this will make 4.0 so much earlier.

File: lib/awful/prompt.lua [hunk 1]
```diff
@@ -461,17 +461,18 @@ function prompt.run(args, textbox, exe_callback, completion_callback,
                     for _,v2 in ipairs(v[1]) do
                         match = match and mod[v2]
                     end
-                    if match or #filtered_modifiers == 0 then
+                    if match then
```
Comment (psychon) [inline lib/awful/prompt.lua]: Is the `or`-part removed on purpose? If so, I don't see why.

The rest looks good to me.

Comment (morethanoneanimal) [inline lib/awful/prompt.lua]: On purpose, it's an impossible condition. If `#filtered_modifiers == 0` then `#v[1] == 0` and we don't enter the previous loop, therefore `local match` remains true.

Comment (psychon) [inline lib/awful/prompt.lua]: Ok, convinced. Would be nicer in a separate commit, but whatever.
